### DISCUSSION TITLE
Inicializar a variável products$ no construtor

### DISCUSSION THREAD

**Discussion by elcioreis** [balta-io/7187#34] [Perguntas e Respostas] [answered]
Olá, pessoal.

Estou fazendo o curso "Criando Apps Web com Angular e UiKit - Curso 7187" 
e acabei parando no seguinte problema:

Pelo que entendi, eu deveria inicializar a variável products$ dentro do construtor.
Como faço essa inicialização?

O ambiente que estou utilizando é:
Angular CLI: 12.1.0
Node: 14.17.1
Package Manager: npm 7.17.0
OS: win32 x64

Muito obrigado pela ajuda.

![image](https://user-images.githubusercontent.com/83420544/124359944-75882e80-dbfd-11eb-8c02-3c0b3db730e4.png)

**Comment by andrebaltieri** [comment, marked as answer]
Boa tarde @elcioreis, você pode inicializar uma propriedade assim:
```ts
public propriedade: Type = VALOR
```

Neste caso, seria algo assim:
```ts
var products$: Observable<any[]> = [];
```

Consegue testar e me dizer se funcionou?

**Reply by elcioreis** [reply]
Botei os colchetes após o símbolo de atribuição e mudou o erro:

![image](https://user-images.githubusercontent.com/83420544/124362816-b4be7b80-dc0d-11eb-91b4-358c5759136c.png)

**Reply by elcioreis** [reply]
Bom dia.

Fiz conforme a dica do colega Rafael Miguel Santos e incluí o item abaixo no arquivo de configuração do projeto tsconfig.json:
"strictPropertyInitialization": false  

Após fazer isso o projeto compila normalmente, mas perco o aviso de erro.

Vou manter essa opção pra seguir com o curso, eu tinha empacado nesse ponto.

**Reply by renatochiari** [reply]
Bom dia,

Aqui resolvi instanciando products$ no construtor:  this.products$ = new Observable();

**Reply by elcioreis** [reply]
Funcionou aqui também.

Obrigado, Renato Chiari.
